File: Ackermann/Haskell/NorfairKing/ackermann.hs
-- The Ackermann function is so easy to define in Haskell.
-- Just write down the definition an you've already almost got working code.
ackermann 0 n = n+1
ackermann m 0 = ackermann (m-1) 1
ackermann m n = ackermann (m-1) (ackermann m (n-1))

File: Ackermann/Python/jcla1/ackermann.py
def ackermann(m, n):
    if m == 0:
        return n+1
    elif m > 0 and n == 0:
        return ackermann(m-1, 1)

    return ackermann(m-1, ackermann(m, n-1))

if __name__ == '__main__':
    print ackermann(3, 5)